store/sqlstore/thread_store.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/store"
+	"github.com/pkg/errors"
+)
+
+type SqlThreadStore struct {
+	SqlStore
+}
+
+func (s *SqlThreadStore) ClearCaches() {
+}
+
+func newSqlThreadStore(sqlStore SqlStore) store.ThreadStore {
+	s := &SqlThreadStore{
+		SqlStore: sqlStore,
+	}
+
+	for _, db := range sqlStore.GetAllConns() {
+		table := db.AddTableWithName(model.Thread{}, "Threads").SetKeys(false, "PostId")
+		table.ColMap("PostId").SetMaxSize(26)
+	}
+
+	return s
+}
+
+func threadSliceColumns() []string {
+	return []string{"PostId", "LastReplyAt", "ReplyCount", "Who"}
+}
+
+func threadToSlice(thread *model.Thread) []interface{} {
+	return []interface{}{
+		thread.PostId,
+		thread.LastReplyAt,
+		thread.ReplyCount,
+		thread.Who,
+	}
+}
+
+func (s *SqlThreadStore) createIndexesIfNotExists() {
+	s.CreateIndexIfNotExists("idx_threads_last_reply_at", "Threads", "LastReplyAt")
+	s.CreateIndexIfNotExists("idx_threads_post_id", "Threads", "PostId")
+}
+
+func (s *SqlThreadStore) SaveMultiple(threads []*model.Thread) ([]*model.Thread, int, error) {
+	builder := s.getQueryBuilder().Insert("Threads").Columns(threadSliceColumns()...)
+	for _, thread := range threads {
+		builder = builder.Values(threadToSlice(thread)...)
+	}
+	query, args, err := builder.ToSql()
+	if err != nil {
+		return nil, -1, errors.Wrap(err, "thread_tosql")
+	}
+
+	if _, err := s.GetMaster().Exec(query, args...); err != nil {
+		return nil, -1, errors.Wrap(err, "failed to save Post")
+	}
+
+	return threads, -1, nil
+}
+
+func (s *SqlThreadStore) Save(thread *model.Thread) (*model.Thread, error) {
+	threads, _, err := s.SaveMultiple([]*model.Thread{thread})
+	if err != nil {
+		return nil, err
+	}
+	return threads[0], nil
+}
+
+func (s *SqlThreadStore) Update(thread *model.Thread) (*model.Thread, error) {
+	if _, err := s.GetMaster().Update(thread); err != nil {
+		return nil, errors.Wrapf(err, "failed to update thread with id=%s", thread.PostId)
+	}
+
+	return thread, nil
+}
+
+func (s *SqlThreadStore) Get(id string) (*model.Thread, error) {
+	var thread model.Thread
+	err := s.GetReplica().SelectOne(&thread, "SELECT * from Threads WHERE PostId=:PostId", map[string]interface{}{"PostId": id})
```

Comment:
This are clearly small enough queries to be written in plain SQL, but maybe we want to move it to squirrel.